### ISSUE TITLE
1.4.6 Answers are Wrong

### ISSUE THREAD

**Issue by VBhojawala** [reneargento/algorithms-sedgewick-wayne#9]
Here is the Correct one

1. O(N)
2.O(N)
3.O(N * logN)

Check out following sites for reference : 

https://stackoverflow.com/questions/29966730/order-of-growth-of-as-function-of-n

https://softwareengineering.stackexchange.com/questions/253421/running-time-of-simple-for-loops

**Comment by reneargento** [comment]
You are correct. I updated the exercise: https://github.com/reneargento/algorithms-sedgewick-wayne/commit/9c1c2a813af39ea4dc24e0e1a2fac31a0f26a241
Thanks!

**Comment by ajfg93** [comment]
@VBhojawala @reneargento Hey, guys, I think the second one is O(2^N), not O(N).
The precise execution count of the inner loop statement `sum++;`  is:

<a href="https://www.codecogs.com/eqnedit.php?latex=2^{N}'&space;-&space;1" target="_blank"><img src="https://latex.codecogs.com/gif.latex?2^{N}'&space;-&space;1" title="2^{N}' - 1" /></a>, 
where N' is the neareset number larger than N, and N' = 2^x, x is integer.

For example, if N = 1000, then N' = 1024, if N = 10, then N' = 2^4 = 18.

**Comment by reneargento** [comment]
Your definition seems to be correct on most of the cases (except when N is a power of 2, check the results below). 
But that still means that the running time grows linearly with the size of the input.

It is easy to see this when running the code and checking the value of `sum`:
```
N = 1 sum = 0
N = 2 sum = 1
N = 4 sum = 3
N = 8 sum = 7
N = 16 sum = 15
N = 32 sum = 31
N = 64 sum = 63
N = 1024 sum = 1023
N = 10 sum = 15
N = 1000 sum = 1023
```
So when N = 16 the inner loop is executed 15 times, which is approximately N. If it were 2^15 that would mean 32768 executions.

**Comment by lucky-dev** [comment]
@ajfg93 About 2nd question.
```
int sum = 0;
for(int i = 1; i < N; i*=2)
  for(int j = 0; j < i; j++)
    sum++
```
Inner loop will execute 1, 2, 4, 8, 16, 32, ... times. We can find sum `1+2+4+8+16+32+...`. It's geometric progression. Use `Sn=b1*(q^n-1)/q-1`. We know `b1 = 1`, `q = 2` and `n = log_2(N) + 1`.
 `log_2(N) + 1` count of members of geometric progression. E.g. for N=32 there will be 6 members. Insert all in formula `2^(log_2(N)+1)-1=2*2^(log_2(N))-1`. Use `a^log_c(b)=b^log_c(a)` and get `2*N^(log_2(2))-1=2*N-1~N`. That's all.
I assume you have found answer (a long time ago). Maybe it will be helpful someone else.

**Comment by ankr2808** [comment]
The why:
N*(1 + 1/2 + .. + 1/N) < 2N
1 + 2 + 4 + .. = 2^lg N - 1 = N - 1 < N
N + N + N + .. = N lg N
The book uses lg = log_2
